### DISCUSSION TITLE
logback appending missing timestamp somehow?

### DISCUSSION THREAD

**Discussion by ScottChapman** [open-telemetry/opentelemetry-java-instrumentation#11543] [Q&A]
So, I am kicking the tires on using OT Java instrumentation and figured I would start with logging. We are using Grafana Alloy as the collector (moral equivalent to OT collector) and Loki for storage.
I am doing automatic instrumentation from my app and using the logback appender to send the logs using OTLP to Alloy. Then Alloy directly to Loki.

In my simple example I noticed that somehow the log messages that make it to Loki lack a timestamp which is puzzling. 

What's the best way to debug this? Is there some easy(ish) way for me to get a console dump of the OTLP leaving my app to see what the data looks like on the way to the collector?

**Comment by laurit** [comment]
> Is there some easy(ish) way for me to get a console dump of the OTLP leaving my app to see what the data looks like on the way to the collector?

Imo the best option is to send data to otel collector and use the debug exporter https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/debugexporter/README.md or logging exporter https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/loggingexporter/README.md
